src/openforms/forms/api/serializers/form_variable.py

```diff
@@ -177,18 +180,47 @@ def validate(self, attrs):
                     }
                 )
 
-        # prefill plugin and attribute must both or both not be set
+        # Check the combination of the provided prefill-attributes (see the model constraints)
+        source = attrs.get("source") or ""
         prefill_plugin = attrs.get("prefill_plugin") or ""
         prefill_attribute = attrs.get("prefill_attribute") or ""
-        if (prefill_plugin and not prefill_attribute) or (
-            not prefill_plugin and prefill_attribute
-        ):
+        prefill_options = attrs.get("prefill_options")
+
+        if prefill_plugin and prefill_options and prefill_attribute:
             raise ValidationError(
                 {
-                    "prefill_attribute": _(
-                        "Prefill plugin and attribute must both be specified."
+                    "prefill_attribute_options": _(
+                        "Prefill plugin, attribute and options can not be specified at the same time."
                     ),
                 }
             )
 
+        if source == FormVariableSources.component:
+            if prefill_options:
+                raise ValidationError(
+                    {
+                        "component_prefill_attribute": _(
+                            "Prefill options should not be specified for component variables."
+                        ),
+                    }
+                )
+            if not prefill_plugin and prefill_attribute:
+                raise ValidationError(
+                    {
+                        "component_prefill_attribute": _(
+                            "Prefill attribute cannot be specified without prefill plugin for component variables."
+                        ),
+                    }
+                )
+
+        # check the specific validation options of the prefill plugin
+        if prefill_plugin:
+            plugin = register[prefill_plugin]
+            serializer = plugin.options(data=prefill_options)
+            try:
+                serializer.is_valid(raise_exception=True)
+            except serializers.ValidationError as e:
+                detail = {"options": e.detail}
```

Comment:
```suggestion
                detail = {"prefill_options": e.detail}
```
needs to map to the serializer field name